src/types/has_at_least.ts

```diff
@@ -0,0 +1 @@
+export type HasAtLeast<T, K extends keyof T> = Partial<T> & Pick<T, K>;
```

Comment:
Am I understanding this type correctly that _any_ prop given as `K` must be present on the object? 
If so, it's bugged, if `T` has a prop typed as `foo?: <type>` it will still be present `type | undefined` on the resulting object

[TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAEghgZwILADIUcAPAFQDRQDSUEAHsBAHYAmCUA1hCAPYBmUOAfFALxQAKcAE7AAlnAA2ubgDIBogMb1cBQpwDcAWABQO0JCgAxZs15QA3jqjWoAI2EAuKAmBDRlAOZbtNu3ABeAPxOlACuALa2EEI6AL46OgD0iVAAcqb60NFCzEIEmVCUcOHQAO7MoRLUUOFwjFCiwM4QEOFQEqL1wAAW0dAI3RVVdtDMlBA6CmMudjmMlE7wyGgYLljGzAQA5Pb+W9x8lj429kJOW0zMW3hxCbraBUso6JgAShAAjqGiQhDUKkQSOQqLQGJd2FwzIIROIpJC5O8vj8-lh+IplPgiJwNHdkmlmKUGsAtnQPCZqJNpk1WKJSH9FohnqtgIjvr9-httrt9mYjr5TudTtc4t4gA)